src/sentry/runner/commands/backup.py

```diff
@@ -59,7 +60,90 @@ def pretty(self) -> str:
+    def check(self, side: str, data: JSONData) -> None:
+        """Ensure that we have received valid JSON data at runtime."""
+
+        if "model" not in data or not isinstance(data["model"], str):
+            raise RuntimeError(f"The {side} input must have a `model` string assigned to it.")
+        if "pk" not in data or not isinstance(data["pk"], int):
+            raise RuntimeError(f"The {side} input must have a numerical `pk` entry.")
+        if "fields" not in data or not isinstance(data["fields"], dict):
+            raise RuntimeError(f"The {side} input must have a `fields` dictionary.")
+
+    @abstractmethod
+    def compare(self, on: InstanceID, left: JSONData, right: JSONData) -> ComparatorFinding | None:
+        """An abstract method signature, to be implemented by inhering classes with their own comparison logic."""
+
+        pass
+
+    def scrub(self, on: InstanceID, left: JSONData, right: JSONData) -> None:
+        """Removes all of the fields compared by this comparator from the `fields` dict, so that the remaining fields may be compared for equality."""
+
+        self.check("left", left)
+        self.check("right", right)
+        if "scrubbed" not in left:
+            left["scrubbed"] = {}
+        if "scrubbed" not in right:
+            right["scrubbed"] = {}
+        for field in self.fields:
+            del left["fields"][field]
+            left["scrubbed"][f"{self.slug()}::{field}"] = True
+            del right["fields"][field]
+            right["scrubbed"][f"{self.slug()}::{field}"] = True
+
+    def slug(self) -> ComparatorSlug:
+        """A unique identifier for this particular derivation of JSONScrubbingComparator, which will be bubbled up in ComparatorFindings when they are generated."""
+
+        return self.__class__.__name__
+
+
+class DateUpdatedComparator(JSONScrubbingComparator):
+    """Comparator that ensures that the specified field's value on the right input is an ISO-8601 date that is greater than (ie, occurs after) the specified field's left input."""
+
+    def __init__(self, field: str):
+        super().__init__([field])
+        self.field = field
+
+    def compare(self, on: InstanceID, left: JSONData, right: JSONData) -> ComparatorFinding | None:
+        left_date_updated = left["fields"][self.field]
+        right_date_updated = right["fields"][self.field]
+        if parser.parse(left_date_updated) > parser.parse(right_date_updated):
+            return ComparatorFinding(
+                name=self.slug(),
+                on=on,
+                reason=f"""the left date_updated value on `{on}` ({left_date_updated}) was not less
+                than or equal to the right ({right_date_updated})""",
+            )
+
+
+ComparatorList = List[JSONScrubbingComparator]
+ComparatorMap = Dict[str, ComparatorList]
+DEFAULT_COMPARATORS: ComparatorMap = {
+    "sentry.userrole": [DateUpdatedComparator("date_updated")],
+    "sentry.userroleuser": [DateUpdatedComparator("date_updated")],
```

Comment:
Yes, every model that needs some special comparison logic will need to list its comparators here. I think eventually we could have this logic live on the models themselves, rather than in this file far away from their definitions, but I don't think that sort of clean up is strictly necessary now.